packages/portal/frontend/src/app/util/UI.ts

```diff
@@ -305,6 +305,34 @@ export class UI {
             Log.error('UI::prompt(..) - ERROR: ' + err.message);
         });
     }
+
+    public static async templateDisplayText(template: string, text: string = ''): Promise<HTMLDivElement> {
+        return ons.createElement(template, {append: true}).then(function(textDialog: any) {
```

Comment:
@rtholmes 

I have implemented the classlist display and the docker build log display in the `UI.ts` file in the last month. The `UI.ts` file is fairly abstract and contains re-usable logic. Can you think of a better place for custom UI code? At the moment, I don't see any examples of custom UI code implemented in other files. If you have any thoughts here, it would be greatly appreciated.